web/src/main/java/se/inera/intyg/webcert/web/service/launchid/InvalidateSessionServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2022 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.webcert.web.service.launchid;
+
+import com.itextpdf.xmp.impl.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.Cache;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.session.FindByIndexNameSessionRepository;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.webcert.web.service.user.dto.WebCertUser;
+import se.inera.intyg.webcert.web.web.controller.api.dto.InvalidateRequest;
+
+
+@Service
+public class InvalidateSessionServiceImpl implements InvalidateSessionService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(InvalidateSessionServiceImpl.class);
+    @Autowired
+    private Cache redisCacheLaunchId;
+    @Autowired
+    private FindByIndexNameSessionRepository<?> sessionRepository;
+
+    @Override
+    public void invalidateSessionIfActive(InvalidateRequest invalidateRequest) {
+        if (launchIdMissing(invalidateRequest.getLaunchId()) || sessionIsMissing(invalidateRequest.getLaunchId())) {
+            LOG.info(String.format("InvalidateSessionServiceImpl called - launchId: %s - has no ongoing session stored in redis.",
+                invalidateRequest.getLaunchId()));
+            return;
+        }
+        final var user = getUserStoredInRedisSession(invalidateRequest.getLaunchId());
```

Comment:
Noticed that we always expect a session to be available in Redis. This is not true as the launchId/session-cache can  be found for a session that has been removed. 

We need to cater for the situation when a session cannot be found.